net/rpc/server.go

```diff
@@ -495,7 +516,18 @@ func (server *Server) ServeRequest(codec ServerCodec) error {
 		}
 		return err
 	}
-	service.call(server, sending, nil, mtype, req, argv, replyv, codec)
+
+	handler :=  func() {
+		service, mtype, req, argv, replyv := service, mtype, req, argv, replyv // capture
```

Comment:
There's no goroutine or for loop here, right? Do we need to capture these variables? As far as I know, the only reason to capture is a for loop or goroutine.

Reply:
hm, I think the closures of the RHS args got me confused here 😕 -- i'll do some digging 📚